CyvoreOS/phone_numbers_extension.py

```diff
@@ -0,0 +1,270 @@
+    )
+    normal = set()
+    for phone in phone_list:
+        logging.debug("Normalizing number: %s", phone)
+        normalized_phone = normalize_phone_number(phone)
+        normal.add(normalized_phone)
+        logging.debug("Normalized number added to the set: %s", normalized_phone)
+
+    normalized_list = list(normal)
+    logging.info(
+        "Normalization complete. Total unique normalized numbers: %d",
+        len(normalized_list),
+    )
+    return normalized_list
+
+
+# ------------------------------------------------------------------------------------------------------
+def validate_number(num: phonenumbers.PhoneNumber) -> str:
+    """
+    Validates a PhoneNumber object from the phonenumbers library.
+    Checks if the number is possible and valid. Returns the formatted number in E.164 format if valid.
+    """
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    try:
+        logging.debug(
+            "Validating phone number: %s",
+            phonenumbers.format_number(num, phonenumbers.PhoneNumberFormat.E164),
+        )
+    except Exception as e:
+        logging.warning("Error formatting number for logging: %s", e)
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    if phonenumbers.is_possible_number(num):
+        logging.debug("The number is a possible number.")
+        if phonenumbers.is_valid_number(num):
+            logging.info("The number is valid.")
+            return phonenumbers.format_number(num, phonenumbers.PhoneNumberFormat.E164)
+        else:
+            logging.warning("The number is possible but not valid.")
+            return "Error - after is_valid_number"
+    else:
+        logging.warning("The number is not possible.")
+        return "Error - after is_possible_number"
+
+
+def retry_with_country_codes(phone_number: str) -> List[str]:
+    """Function will return a list of the same number with different country codes that validate successfully.
+    Special handling is applied for countries that retain leading zeros in phone numbers.
+    """
+    logging.info("Starting retry with country codes for number: %s", phone_number)
+    valid_numbers = []
+    stripped_from_plus = phone_number.lstrip("+")
+
+    for code in country_codes:
+        try:
+            if code not in special_retain_zero_country_codes:
+                full_number = code + stripped_from_plus.lstrip("0")
+                logging.debug("Trying with country code %s: %s", code, full_number)
+            else:
+                full_number = code + stripped_from_plus  # keep '0'
+                logging.debug(
+                    "< special_retain_zero > Trying country code %s with leading '0' for number: %s",
+                    code,
+                    full_number,
+                )  # ["+39", "+33", "+46"]
+            parsed_number = phonenumbers.parse(full_number)
+            result = validate_number(parsed_number)
+
+            if "Error" not in result and result not in valid_numbers:
```

Comment:
After adding try except this scope if should be inside the except and no more need in the if statement